include/zephyr/toolchain/common.h

```diff
@@ -160,10 +175,10 @@
  */
 
 /* Indicates this is a driver subsystem */
-#define __subsystem
+#define __subsystem __zpp(struct("__subsystem"))
 
 /* Indicates this is a network socket object */
-#define __net_socket
+#define __net_socket __zpp(struct("__net_socket"))
```

Comment:
for future: these should likely be moved somewhere else

Reply:
Agreed